### PR TITLE
GH-2596 - Claim Tracking

### PR DESCRIPTION
**Pull request by biafra23**
Handles #2596 

Changes proposed in this pull request:
- Add tracking ids
- Add tracking

### CODE REVIEW

File: Multisig/Cross-layer/Analytics/Tracker.swift [hunk 1]
```diff
@@ -147,6 +147,7 @@ protocol ScreenTrackingEvent: Trackable {}
 
 extension Tracker {
     static func trackEvent(_ event: TrackingEvent, parameters: [String: Any]? = nil, file: StaticString = #file, line: UInt = #line, function: StaticString = #function) {
+        print("------------> trackEvent(\(event))")
```
Comment (biafra23) [inline Multisig/Cross-layer/Analytics/Tracker.swift]: Note to self: Don't forget to delete this.

File: Multisig/UI/ClaimToken/ClaimingAmountViewController/ClaimTokensViewController.swift [hunk 2]
```diff
@@ -201,12 +190,15 @@ class ClaimTokensViewController: LoadableViewController {
...
     @IBAction private func editButtonTouched(_ sender: Any) {
+        Tracker.trackEvent(.userClaimFormDel)
         onEditDelegate()
     }
 
     // pull-to-refresh, initial reload
     override func reloadData() {
         super.reloadData()
+        //TODO Do not track for initial load.
+        Tracker.trackEvent(.userClaimFormReload)
```
Comment (DmitryBespalov) [inline Multisig/UI/ClaimToken/ClaimingAmountViewController/ClaimTokensViewController.swift]: put it to a override of the `didStartRefreshing()` method

File: Multisig/UI/ClaimToken/Select Guardian/GuardianListViewController.swift [hunk 3]
```diff
@@ -151,6 +144,10 @@ extension GuardianListViewController: UISearchResultsUpdating, UISearchBarDelega
     }
 
     func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+
+        // TDOO  Fix: Hits only when Enter is pressed in search bar
+        Tracker.trackEvent(.userClaimChdelSearch)
```
Comment (DmitryBespalov) [inline Multisig/UI/ClaimToken/Select Guardian/GuardianListViewController.swift]: just put it at the end of the `updateSearchResults` method

Comment (DmitryBespalov) [reply Multisig/UI/ClaimToken/Select Guardian/GuardianListViewController.swift]: Or implement a [searchBarTextDidBeginEditing(_:)](https://developer.apple.com/documentation/uikit/uisearchbardelegate)

**Review by DmitryBespalov — COMMENTED** [review]
epic! 👏 few more things and I think it's good

File: Multisig/UI/ClaimToken/Select Guardian/GuardianListViewController.swift [hunk 4]
```diff
@@ -148,12 +141,12 @@ extension GuardianListViewController: UISearchResultsUpdating, UISearchBarDelega
         }
         resultsController.selectedDelegate = selectedDelegate
         resultsController.tableView.reloadData()
+        Tracker.trackEvent(.userClaimChdelSearch)
```
Comment (DmitryBespalov) [inline Multisig/UI/ClaimToken/Select Guardian/GuardianListViewController.swift]: here you can switch between this event and `screenClaimChdelNf`:
```suggestion
        Tracker.trackEvent(resultsController.filteredGuardians.isEmpty ? .screenClaimChdelNf : .userClaimChdelSearch)
```

File: Multisig/Cross-layer/Analytics/TrackingEvent.swift [hunk 5]
```diff
@@ -329,9 +329,86 @@ enum TrackingEvent: String, Trackable {
...
+    // Legal disclaimer screen Tracking
+    case screenClaimLegal                           = "screen_claim_legal"
+    case userClaimLegalAgree                        = "user_claim_legal_agree"
+
+    // Choose a delegate screen Tracking
+    case screenClaimChdel                           = "screen_claim_chdel"
+    case userClaimChdelGuard                        = "user_claim_chdel_guard"
+    case userClaimChdelAddr                         = "user_claim_chdel_addr"
+    case userClaimChdelSearch                       = "user_claim_chdel_search"
+    case screenClaimChdelNf                         = "screen_claim_chdel_nf" // not used
```
Comment (DmitryBespalov) [inline Multisig/Cross-layer/Analytics/TrackingEvent.swift]: comment can be removed

**Review by DmitryBespalov — APPROVED** [review]

File: Multisig/UI/ClaimToken/Select Guardian/GuardianListViewController.swift (hunk 4 above)
Comment (biafra23) [reply Multisig/UI/ClaimToken/Select Guardian/GuardianListViewController.swift]: We do want to track `userClaimChdelSearch` even if there is no result, nor?

**Pull request merged**